src/nupic/algorithms/Connections.cpp

```diff
@@ -89,7 +80,7 @@ Synapse Connections::createSynapse(const Segment& segment,
                                    Permanence permanence)
 {
   vector<SynapseData>& synapses = cells_[segment.cell.idx].segments[segment.idx].synapses;
-  if (synapses.size() == SYNAPSE_MAX)
+  if (synapses.size() == MAX_SYNAPSES_PER_SEGMENT)
```

Comment:
That would introduce another variable to serialize and deserialize and test. It's not something the client can change yet, and I wanted to keep that change in another PR. I could add a TODO and file an issue now; or I could just implement that feature in this PR. What do you suggest?